src/components/DataTable/DataTable.js

```diff
@@ -389,7 +389,7 @@ export default class DataTable extends React.Component {
           return p;
         }, {});
         return {
-          shouldShowBatchActions: false,
+          shouldShowBatchActions: !row.isSelected || selectedRowsCount > 0,
```

Comment:
Seems that what `handleOnSelectRow()` returns is almost identical between radio/checkbox with your change. Would you be able to try reducing the duplicated code? Also it appears that the code tries to access `selectedRowsCount` before it's defined.

Also, just found that the existing code (above `reduce()` logic) seems to mutate an object, is it true? If so, we should create a clone with the new `isSelected` prop given the nature of React (where its change detection logic relies on immutable objects).